### PR TITLE
BookShortcuts plugin: fix folders handling

### PR DESCRIPTION
**Pull request by hius07**
Fixes:

(1)
`02/09/22-16:06:51 WARN  FileManager instance mismatch! Tried to spin up a new instance, while we still have an existing one: table: 0x42d62bb8
`

(2)
`02/09/22-16:06:51 ERROR failed to call event handler onBookShortcut frontend/apps/reader/readerui.lua:545: bad argument #1 to 'attributes' (string expected, got nil)
`

(3) `Directory` to `Folder`

---
This change is [Reviewable](https://reviewable.io/reviews/koreader/koreader/8795)

### CODE REVIEW

File: plugins/bookshortcuts.koplugin/main.lua [hunk 1]
```diff
@@ -36,16 +36,27 @@ function BookShortcuts:onBookShortcut(path)
...
             if G_reader_settings:readSetting("BookShortcuts_directory_action") == "FM" then
-                local FileManager = require("apps/filemanager/filemanager")
-                FileManager:showFiles(path)
+                if self.ui.file_chooser then
+                    self.ui.file_chooser:changeToPath(path)
+                else -- called from Reader
+                    self.ui:onClose()
+                    local FileManager = require("apps/filemanager/filemanager")
+                    if FileManager.instance then
+                        FileManager.instance:reinit(path)
```
[the remark this answers is not in the document]
Comment (hius07) [reply plugins/bookshortcuts.koplugin/main.lua:45]: In all my test cases FM.instance did not exist.
This code was introduced in https://github.com/koreader/koreader/pull/4759.

[the remark this answers is not in the document]
Comment (NiLuJe) [reply plugins/bookshortcuts.koplugin/main.lua:45]: FileManager itself *should* though. The extra precautions are taken for reader.lua's sake where some stuff might be called as the first thing we do on startup.

**Comment by hius07** [comment]
https://github.com/koreader/koreader/blob/4ccdbc90a573579d626af47908440d3914960427/frontend/apps/filemanager/filemanagershortcuts.lua#L41-L45
https://github.com/koreader/koreader/blob/4ccdbc90a573579d626af47908440d3914960427/frontend/apps/filemanager/filemanagerfilesearcher.lua#L215-L219
This piece of code is duplicated in several modules.
Can it be included into
https://github.com/koreader/koreader/blob/4ccdbc90a573579d626af47908440d3914960427/frontend/apps/filemanager/filemanager.lua#L1262-L1268
Do not close existing FM.instance, but reinit?

**Comment by poire-z** [comment]
^ Comes from #7658.
Not that familiar with how/why all this to give you any go/nogo :/

**Comment by NiLuJe** [comment]
c.f., https://github.com/koreader/koreader/pull/8795#discussion_r803604549

No time to check right now  if that applies here as well ;).

**Comment by NiLuJe** [comment]
Oops, that was another line of questioning ;p.

`reinit` reuses the existing instance, so it's very goal is to avoid having to destroy/create another one ;).

**Comment by hius07** [comment]
Thank you, noted.

**Review by yparitcher — APPROVED** [review]
Looks good.

When i wrote this i saw
```lua
--- @note: This is the *only* safe way to instantiate a new FileManager instance!
function FileManager:showFiles(path, focused_file)
```
so i assumed it took care of closing everthing. I was wrong :)

Should fix #8762

**Comment by NiLuJe** [comment]
> so i assumed it took care of closing everthing. I was wrong :)

It technically does, it just shouts loudly about having to do it ;).

**Comment by yparitcher** [comment]
It does not properly close 
> It technically does, it just shouts loudly about having to do it ;).
> 
It does not properly close the Reader instance with a `self.ui:onClose()` leading to unsaved settings and other mischeif

**Comment by NiLuJe** [comment]
Oh, right, I assumed we were only talking about FM ;o).

**Pull request merged**